jax/random.py

```diff
@@ -417,19 +418,66 @@ def _randint(key, shape, minval, maxval, dtype):
   return lax.add(minval, lax.convert_element_type(random_offset, dtype))
 
 
-def shuffle(key: np.ndarray, x: np.ndarray, axis: int = 0) -> np.ndarray:
-  """Shuffle the elements of an array uniformly at random along an axis.
+def shuffle(key: np.ndarray, x: np.ndarray, axis: Optional[int]=0) -> np.ndarray:
+  """Shuffle the elements of an array uniformly along the first axis.
 
   Args:
     key: a PRNGKey used as the random key.
     x: the array to be shuffled.
-    axis: optional, an int axis along which to shuffle (default 0).
+    axis: (deprecated) an int axis along which to shuffle (default 0). If not None,
+      subarrays are shuffled independently along the specified axis, as in
+      :func:`~jax.random.shuffle_axis`. If `None`, subarrays are shuffled as a group.
+      along the leading axis, as in :func:`numpy.random.shuffle`.
+      In a future release, the default will be changed to `None` and support for
+      integer axis values will be dropped.
```

Comment:
Long term, do we need support for shuffling in groups at all?

- NumPy seems to get along without it OK, just creating explicit arrays to use with indexing
- We have `vmap`, so arguably the idiomatic way to shuffle a bunch of arrays together is to `vmap` but not batch the key.